### PR TITLE
adding keyboard: pain27

### PR DESCRIPTION
**Pull request by worldspawn00**
## Description

adding keyboard: pain27
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/wsk/pain27/rules.mk [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration
```

File: keyboards/wsk/pain27/rules.mk [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
MOUSEKEY_ENABLE ?= yes	# Mouse keys
```

File: keyboards/wsk/pain27/rules.mk [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
EXTRAKEY_ENABLE ?= yes	# Audio control and System control
```

File: keyboards/wsk/pain27/rules.mk [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
CONSOLE_ENABLE ?= no	# Console for debug
```

File: keyboards/wsk/pain27/rules.mk [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
COMMAND_ENABLE = no    # Commands for debug and configuration
```

File: keyboards/wsk/pain27/rules.mk [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
```

File: keyboards/wsk/pain27/rules.mk [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```

File: keyboards/wsk/pain27/rules.mk [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
```

File: keyboards/wsk/pain27/rules.mk [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+AUDIO_ENABLE ?= no
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
AUDIO_ENABLE = no
```

File: keyboards/wsk/pain27/rules.mk [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+RGBLIGHT_ENABLE ?= yes
```
Comment (ridingqwerty) [inline keyboards/wsk/pain27/rules.mk]:
```suggestion
RGBLIGHT_ENABLE = yes
```

File: keyboards/wsk/pain27/config.h [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
```
Comment (drashna) [inline keyboards/wsk/pain27/config.h]: This is set to the default, and should be removed.

File: keyboards/wsk/pain27/readme.md [hunk 12]
```diff
@@ -0,0 +1,16 @@
+# Pain27v2
+
+![Pain27v2](https://i.imgur.com/qJZlelV.png)  
+
+The Pain27v2 is an adaptation of the original Pain27 board by jonathan(uuupah), adding RGB LEDs and moving the controller under the keys.  
+
+Keyboard Maintainer: [worldspawn00](https://github.com/worldspawn00/)
```
Comment (drashna) [inline keyboards/wsk/pain27/readme.md]:
```suggestion
* Keyboard Maintainer: [worldspawn00](https://github.com/worldspawn00/)  
```

File: keyboards/wsk/pain27/readme.md [hunk 13]
```diff
@@ -0,0 +1,16 @@
+# Pain27v2
+
+![Pain27v2](https://i.imgur.com/qJZlelV.png)  
+
+The Pain27v2 is an adaptation of the original Pain27 board by jonathan(uuupah), adding RGB LEDs and moving the controller under the keys.  
+
+Keyboard Maintainer: [worldspawn00](https://github.com/worldspawn00/)  
+Hardware Supported: The Pain27v2 PCB and kit.
```
Comment (drashna) [inline keyboards/wsk/pain27/readme.md]:
```suggestion
* Hardware Supported: The Pain27v2 PCB and kit.  
```

File: keyboards/wsk/pain27/readme.md [hunk 14]
```diff
@@ -0,0 +1,16 @@
+# Pain27v2
+
+![Pain27v2](https://i.imgur.com/qJZlelV.png)  
+
+The Pain27v2 is an adaptation of the original Pain27 board by jonathan(uuupah), adding RGB LEDs and moving the controller under the keys.  
+
+Keyboard Maintainer: [worldspawn00](https://github.com/worldspawn00/)  
+Hardware Supported: The Pain27v2 PCB and kit.  
+Hardware Availability: [PCB & Kit](https://geekhack.org/index.php?topic=100283)
```
Comment (drashna) [inline keyboards/wsk/pain27/readme.md]:
```suggestion
* Hardware Availability: [PCB & Kit](https://geekhack.org/index.php?topic=100283)  
```

**Comment by ridingqwerty** [comment]
Committed suggestions made by @drashna and myself.

**Pull request merged**